cirq/protocols/approx_equality.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from typing import Any
+
+from typing_extensions import Protocol
+
+
+class SupportsApproxEquality(Protocol):
+    """Object which can be approximately compared."""
+
+    def _approx_eq_(self, other: Any, rel_tol: float, abs_tol: float) -> bool:
```

Comment:
Force keyword args when setting the tolerances by adding a `, *,` argument after the `other` parameter.

Reply:
Added `, *` for all the internal methods. I keep `approx_eq()` without forcing keyword args?

Reply:
Done.